backend/server.js

```diff
@@ -1,39 +1,78 @@
 if (process.env.NODE_ENV !== 'production') { // can load .env file while server isn't in production mode
     require('dotenv').config();
 }
-
-const authHelper = require('./authHelper')
-const prisma = require('./database/db')
-const express = require('express') 
-const app = express()
-const port = 8080
+const express = require('express'); 
+const app = express();
+const cors = require('cors');
 const bcrypt = require('bcrypt')
-const session = require('express-session')
-const {PrismaSessionStore} = require('@quixo3/prisma-session-store')
-const {PrismaClient} = require('@prisma/client')
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
+const port = 8080;
+const {authenticateSession, userSession, userSessionRouter} = require('./routes/userSession.js');
 const usernameRouter = require('./routes/usernameRoute');
-app.use(cors({ origin:'http://localhost:3000'}));
-app.use('/reset-username-form', usernameRouter)
+const authHelper = require('./authHelper')
 
 app.use(express.json()); 
-app.use(session({
-    secret: process.env.SESSION_SECRET,
-    resave: false,
-    saveUninitialized: true,
-    cookie: {secure: true},
-    store: new PrismaSessionStore(
-        new PrismaClient(),
-        {
-            checkPeriod: 2*60*1000,
-            dbRecordIdIsSessionId: true,
-            dbRecordIdFunction: undefined,
+app.use(cors({
+    origin: 'http://localhost:3000/',
+    credentials: true
+  }));
+app.use('/reset-username-form', usernameRouter)
+app.use('/user-session', userSessionRouter)
+app.use(userSession)
+
+app.post('/sign-in', async (request, response) => { // Sign in route, get the users data to login
```

Comment:
extract the route handler function elsewhere so you can call it by name and try simplifying the logic. There is a ton of try catches and conditionals and there is probably some logic that is duplicate/ not useful which you can remove.